snapshot-labs/sx-monorepo
# feat: integrate Axiom

### Summary

Closes: https://github.com/snapshot-labs/sx-monorepo/issues/156
Staging branch on: https://github.com/snapshot-labs/sx-monorepo/pull/4

This PR adds support for Axiom execution strategy (this one is mock from Axiom team).

When creating space you can now add Axiom execution strategy. With it enabled votes will be broadcast to highlight instance and later proven by Axiom.

Once voting window ends you can generate proofs from UI and execute transaction as-if votes were on-chain.

### Test plan

- Run `yarn dev:full`, you can change sepolia network's start index.
- Mint & delegate to yourself this token https://sepolia.etherscan.io/address/0x6fd821e79cdf212ad8b06c59b28fe8c2185291d4#writeContract
- Create space on Sepolia with ERC-20 Comp tokens voting/proposition power strategies using this contract `0x6fd821e79cdf212ad8b06c59b28fe8c2185291d4`.
- Use Axiom execution strategy (contract `0x6fd821e79cdf212ad8b06c59b28fe8c2185291d4`, slotIndex `8`).
- Change max voting delay to something shorter (10 minutes).
- After space is deployed go to settings and copy execution address, add it as treasury.
- Send some ERC20 token to this treasury to use as execution.
- Create proposal **with axiom execution** and vote on it.
- Once voting period has ended you should see `Finalize proposal` button, click on it and wait some time (5-10 minutes).
- Refresh page, it should show `Execute proposal` eventually, click this button.
- Wait some time, refresh page, it should show link to proposal execution.

Alternatively use this space instead of creating your own: http://localhost:8080/#/sep:0x648b2C6710f92d59E2562a61E9bdF2Cbe8f87186
